### PR TITLE
Add symbols API

### LINKED ISSUE #15: Implement symbols
It should be possible to expose a decent cross object symbol api.

### PR DESCRIPTION
**Pull request by philipc**
There's sure to be lots missing from this, but it's enough to match most of the nm and llvm-nm output for some ELF and Mach-O files, and should be sufficient for addr2line (but I haven't done that yet).

Fixes #15

### CODE REVIEW

**Comment by philipc** [comment]
I've also got this working with addr2line now.

**Comment by fitzgen** [comment]
Is this ready for a review?

**Comment by philipc** [comment]
Yes please.

File: src/lib.rs [hunk 1]
```diff
@@ -111,6 +273,118 @@ fn macho_translate_section_name(section_name: &str) -> Vec<u8> {
...
+                section_kinds.push((SectionKind::Unknown, SymbolKind::Unknown));
+            }
+        }
+    }
+
+    let mut symbols = Vec::new();
+    for sym in macho.symbols() {
+        if let Ok((name, nlist)) = sym {
+            // Skip STAB debugging symbols.
+            if nlist.n_type & 0xe0 != 0 {
```
Comment (fitzgen) [inline src/lib.rs]: What is this magic constant? Would be good to name it.

Comment (philipc) [reply src/lib.rs]: N_STAB, I need to add it to goblin, but I added a comment here for now.

**Pull request merged**